frameworks/web-ui/react/src/App.tsx

```diff
@@ -1,33 +1,59 @@
 import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
 import './App.css'
+import { Button } from './Button'
+
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [equation, setEquation] = useState<string>("0");
+
+  const addToEquation = (token: string) => {
+    setEquation((equation) => {
+      if (equation == "0") {
+        return token;
+      }
+      return equation + token
+    });
+  }
+
+  const clear = () => {
+    setEquation((equation) => "0");
+  }
+
+  const calculate = () => {
+    setEquation((equation) => eval(equation));
+  }
 
+  //TODO: What I would really like is to pass a closure to 
+  //the Button component that it would call without any extra params
+  //so that I could pass in `calculate` and `clear` the same as
+  //addToEquation
   return (
     <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
+      <h1>Calculator</h1>
       <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
+        <h2>{equation}</h2>
+
+        <br/>
+        <Button text="1" clickFunction={addToEquation} />
+        <Button text="2" clickFunction={addToEquation}/>
+        <Button text="3" clickFunction={addToEquation} />
+        <br/>
+        <Button text="4" clickFunction={addToEquation} />
+        <Button text="5" clickFunction={addToEquation} />
+        <Button text="6" clickFunction={addToEquation} />
+        <br/>
+        <Button text="7" clickFunction={addToEquation} />
+        <Button text="8" clickFunction={addToEquation} />
+        <Button text="9" clickFunction={addToEquation} />
+        <br/>
+        <Button text="0" clickFunction={addToEquation} />
+        <Button text="+" clickFunction={addToEquation} />
+        <Button text="-" clickFunction={addToEquation} />
+        <br/>
+        <button onClick={() => calculate()} >=</button>
```

Comment:
You can actually just write this as: `<button onClick={calculate}>=</button>`
The same is applicable as below.